AdobeStockImage/Test/Integration/Model/SaveImageTest.php

```diff
@@ -0,0 +1,219 @@
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $criteriaBuilder;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp(): void
+    {
+        $this->driver = Bootstrap::getObjectManager()->get(DriverInterface::class);
+        $this->fileSystem = Bootstrap::getObjectManager()->get(Filesystem::class);
+        $this->assetRepository = Bootstrap::getObjectManager()->get(AssetRepositoryInterface::class);
+        $this->criteriaBuilder = Bootstrap::getObjectManager()->get(SearchCriteriaBuilder::class);
+
+        $this->deleteImage();
+        $https = $this->createMock(Https::class);
+        $https->expects($this->once())
+            ->method('fileGetContents')
+            ->willReturnCallback(function ($filePath) {
+                return file_get_contents($filePath);
+            });
+        $storageSave = Bootstrap::getObjectManager()->create(Save::class, ['driver' => $https]);
+        $saveImageFile = Bootstrap::getObjectManager()->create(SaveImageFile::class, ['storageSave' => $storageSave]);
+        $this->saveImage = Bootstrap::getObjectManager()->create(
+            SaveImageInterface::class,
+            ['saveImageFile' => $saveImageFile]
+        );
+    }
+
+    /**
+     * @inheridoc
+     */
+    protected function tearDown(): void
+    {
+        $this->deleteImage();
+        parent::tearDown();
+    }
+
+    /**
+     * Test with image.
+     *
+     * @return void
+     */
+    public function testSave(): void
+    {
+        $document = $this->getDocument();
+        $this->saveImage->execute(
+            $document,
+            $document->getCustomAttribute(self::URL_FIELD)->getValue(),
+            $this->fileSystem->getDirectoryWrite(DirectoryList::MEDIA)->getAbsolutePath($this->saveDestination)
+        );
+        self::assertTrue(
+            $this->fileSystem->getDirectoryRead(DirectoryList::MEDIA)->isExist($this->saveDestination),
+            'File was not saved by destination'
+        );
+        $searchCriteria = $this->criteriaBuilder
+            ->addFilter('creator_id', $document->getCustomAttribute('creator_id')->getValue(), 'eq')
+            ->create();
+        self::assertNotEmpty(
+            $this->assetRepository->getList($searchCriteria),
+            'Image asset was not saved'
+        );
+    }
+
+    /**
+     * Document for save.
+     *
+     * @return Document
+     * @throws IntegrationException
+     */
+    private function getDocument(): Document
+    {
+        $stockFileData = [
+            'id' => 1,
+            'comp_url' => 'https://test.url/1.png',
+            'width' => 110,
+            'title' => 'test',
+            'content_type' => 'image/png',
+            'height' => 210,
+            'some_bool_param' => false,
+            'some_nullable_param' => null,
+            'category' => [
+                'id' => 1,
+                'name' => 'Test'
+            ],
+        ];
+
+        $stockFile = new StockFile($stockFileData);
```

Comment:
I don't see the reason to create a stock file first and then convert it to document for this test. Please create a document straight away